### PR TITLE
hmem: Implement synapseai iface

### PR DESCRIPTION
**Pull request by shijin-aws**
This PR implements the required configure flags and hmem ops for synapseai hmem iface.

### CODE REVIEW

File: src/hmem_synapseai.c [hunk 1]
```diff
@@ -37,6 +37,155 @@
...
+{
+	dlclose(synapseai_handle);
+}
+
+/**
+ * @brief Initialize SynapseAI hmem interface by dlopen the SynapseAI library
+ * 
+ * @return On success, return 0. On failure, return negative error code.
+ */
+int synapseai_init(void)
```
Comment (wzamazon) [inline src/hmem_synapseai.c:94]: It looks like the only thing `synapseai_init` does was to call synapseai_dl_init`. Can we combine them into one?

File: src/hmem_synapseai.c [hunk 2]
```diff
@@ -37,6 +37,155 @@
...
+		return ret;
+
+	return FI_SUCCESS;
+}
+
+/**
+ * @brief Clean up SynapseAI intereface (dlclose)
+ * @return return 0. 
+ */
+int synapseai_cleanup(void)
```
Comment (wzamazon) [inline src/hmem_synapseai.c:109]: Similarly, can we combine `synapseai_cleanup` and `synapseai_dl_cleanup`

**Review by wzamazon — CHANGES_REQUESTED** [review]
Two minor comments.

File: src/hmem_synapseai.c (hunk 1 above)
Comment (shijin-aws) [reply src/hmem_synapseai.c]: Fixed.

File: src/hmem_synapseai.c (hunk 2 above)
Comment (shijin-aws) [reply src/hmem_synapseai.c:84]: Done

**Review by wzamazon — APPROVED** [review]

**Review by shefty — APPROVED** [review]

**Comment by shijin-aws** [comment]
AWS CI failed due to ssh connection issue .. retesting.

**Comment by shijin-aws** [comment]
bot:aws:retest

**Pull request merged**